File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
    int no_of_digits = floor(log10f(candidate)) + 1;
    int sum_of_pows = 0;

    for (int i = 0; i < no_of_digits; i++) {
        int digit = (candidate % (int)pow(10, i + 1)) / pow(10, i);
        sum_of_pows += pow(digit, no_of_digits);
    }

    return sum_of_pows == candidate;
}

File: go/armstrong-numbers/armstrong_numbers.go
package armstrongnumbers

import (
	"math"
)

func IsNumber(n int) bool {
	numOfDigits := int(math.Floor(math.Log10(float64(n)))) + 1
	sumOfPows := 0

	for i := range numOfDigits {
		digit := (n % int(math.Pow(10, float64(i+1)))) / int(math.Pow(10, float64(i)))
		sumOfPows += int(math.Pow(float64(digit), float64(numOfDigits)))
	}

	return sumOfPows == n
}
